src/js/libcs/Accessors.js

```diff
@@ -168,6 +168,24 @@ Accessor.setField = function(geo, field, value) {
         movepointscr(geo, List.turnIntoCSList([value.value[0], value.value[1], CSNumber.real(1)]), "homog");
     }
 
+    if (field === "x" && geo.kind === "P" && geo.movable && (List._helper.isNumberVecN(value, 1) || value.ctype === "number")) {
+        if (CSNumber._helper.isAlmostZero(geo.homog.value[2])) {
+            console.log("cannot set x component without y component of far points");
```

Comment:
By the way, do we `normalizeMax` on assignment? This would make change of representant impossible. (I only see pathological use cases for change of representant, anyway.)